### PR TITLE
Add spec_id, partition, sort_order_id, readable_metrics columns to Iceberg $files table

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Relates to #24101

Add 4 missing columns Spark supports: https://iceberg.apache.org/docs/latest/spark-queries/#files
* spec_id: INTEGER
* partition: ROW(...)
* sort_order_id: INTEGER
* readable_metrics: JSON

## Release notes

```markdown
## Iceberg
* Add `spec_id`, `partition`, `sort_order_id`, `readable_metrics` columns to `$files` metadata table. ({issue}`24102`)
```

### CODE REVIEW

**Comment by osscm** [comment]
@ebyhr thanks for adding this feature!!

Can you please take my changes/commit and add yours on top of it. 
If you see, I was waiting for that PR to be review from long time. :(

thanks

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java
Comment (sopel39) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java:411]: this could potentially be sensitive information which might be guarded by access control rules

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java
Comment (sopel39) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java:1144]: unused remove

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergSystemTables.java
Comment (sopel39) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergSystemTables.java:482]: what happens when you rename part column (for new and old file)?

**Review by sopel39 — APPROVED** [review]

File: docs/src/main/sphinx/connector/iceberg.md
Comment (ebyhr) [reply docs/src/main/sphinx/connector/iceberg.md:1295]: It contains column size, value count, null & nan value counts, lower & upper bound. 
It can be sensitive information, but the table already has id-based lower & upper bound columns.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergSystemTables.java
Comment (ebyhr) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergSystemTables.java]: It shows the original column names. That is the Spark behavior:
```sql
spark-sql (default)> CREATE TABLE test(x int, part int) USING iceberg PARTITIONED BY (part);
spark-sql (default)> INSERT INTO test VALUES (1, 10);
spark-sql (default)> ALTER TABLE test RENAME COLUMN part to renamed;
spark-sql (default)> SELECT partition FROM iceberg_test.default.test.files;
{"part":10}
spark-sql (default)> INSERT INTO test VALUES (2, 20);
spark-sql (default)> SELECT partition FROM iceberg_test.default.test.files;
{"part":20}
{"part":10}
```

**Pull request merged**
